File: backend/semgrep_rules/tests/python/kisa_sw_43_unsafe_deserialization.py
import pickle
def unsafe(request):
    data = request.POST.get("data")
    # ruleid: kisa.sw43.python.external-input-to-deserializer
    return pickle.loads(data)
def safe(request):
    import json
    data = request.POST.get("data")
    # ok: kisa.sw43.python.external-input-to-deserializer
    return json.loads(data)

File: backend/semgrep_rules/tests/javascript/kisa_sw_43_unsafe_deserialization.js
const serialize = require("node-serialize");
function unsafe(req) {
  const data = req.body.data;
  // ruleid: kisa.sw43.javascript.external-input-to-unsafe-deserializer
  return serialize.unserialize(data);
}
function safe(req) {
  const data = req.body.data;
  // ok: kisa.sw43.javascript.external-input-to-unsafe-deserializer
  return JSON.parse(data);
}
